dairy-ns/src/modules/milk-report/milk-report.service.ts

```diff
@@ -0,0 +1,107 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model, PaginateModel, PaginateResult } from 'mongoose';
+import {
+  CreateMilkReportDto,
+  UpdateMilkReportDto,
+} from 'src/constants/dto/milk-report.dto';
+import { MilkReport } from 'src/schema/milk-report/milk-report.schema';
+// import { UpdateMilkReportDto } from './dto/update-milk-report.dto';
+
+@Injectable()
+export class MilkReportService {
+  constructor(
+    @InjectModel(MilkReport.name)
+    private milkReportModel: PaginateModel<MilkReport>,
+  ) {}
+
+  create(createMilkReportDto: CreateMilkReportDto, filledBy: string) {
+    return this.milkReportModel.create({ ...createMilkReportDto, filledBy });
+  }
+
+  async findAll(
+    of: string,
+    byMe?: boolean,
+    page: number = 1,
+    perPage: number = 10,
+    from?: string,
+    to?: string,
+    milkType?: string,
+    isRejected?: boolean,
+  ): Promise<PaginateResult<MilkReport>> {
```

Comment:
No need for these many variables, just use two variables

from, and limit

Reply:
actually this variables are for filterinf 

> No need for these many variables, just use two variables
> 
> from, and limit

actually this variable are for filtering purpose , if inspector want to fetch reports in between date range or according to milkType , they are serving that purpose with this single api